gateway/ln-gateway/src/bin/gatewayd.rs

```diff
@@ -39,23 +41,29 @@ pub struct GatewayOpts {
     /// Gateway webserver authentication password
     #[arg(long = "password", env = "FM_GATEWAY_PASSWORD")]
     pub password: String,
+}
 
-    /// Public URL to a Gateway Lightning rpc service
-    #[arg(long = "lnrpc-addr", env = "FM_GATEWAY_LIGHTNING_ADDR")]
-    pub lnrpc_addr: Option<Url>,
-
-    // TODO: Issue #1955: Group clap args for configuring gatewayd with LND
-    /// LND RPC address
-    #[arg(long = "lnd-rpc-host", env = "FM_LND_RPC_ADDR")]
-    pub lnd_rpc_addr: Option<String>,
-
-    /// LND TLS cert file path
-    #[arg(long = "lnd-tls-cert", env = "FM_LND_TLS_CERT")]
-    pub lnd_tls_cert: Option<String>,
-
-    /// LND macaroon file path
-    #[arg(long = "lnd-macaroon", env = "FM_LND_MACAROON")]
-    pub lnd_macaroon: Option<String>,
+#[derive(Debug, Clone, Subcommand)]
+enum Mode {
+    #[clap(name = "LND")]
+    Lnd {
+        /// LND RPC address
+        #[arg(long = "lnd-rpc-host", env = "FM_LND_RPC_ADDR")]
+        lnd_rpc_addr: String,
+
+        /// LND TLS cert file path
+        #[arg(long = "lnd-tls-cert", env = "FM_LND_TLS_CERT")]
+        lnd_tls_cert: String,
+
+        /// LND macaroon file path
+        #[arg(long = "lnd-macaroon", env = "FM_LND_MACAROON")]
+        lnd_macaroon: String,
+    },
+    #[clap(name = "CLN")]
+    Cln {
+        #[arg(long = "lnrpc-addr", env = "FM_GATEWAY_LIGHTNING_ADDR")]
+        lnrpc_addr: Url,
```

Comment:
Can we rename this to extension address or something like that? Otherwise it's easy to confuse with the unix socket CLN uses.

Reply:
Agreed, done.